app/controllers/application_controller.rb

```diff
@@ -198,8 +214,7 @@ def redirect_back_or(path)
 
   def redirect_to_siginin_if_anon_or(path)
     if current_user.nil?
-      session[:redirect_path_after_signin] = request.path
-      redirect_to new_user_session_path
+      redirect_to auth_login_path(after_sign_in_path: request.path)
     elsif !path.empty?
```

Comment:
I'm wondering if we should have a fallback to the root if the path is empty.  Based on the current code paths path shouldn't be empty but if `after_sign_in_path_for(current_user)` ever returns null the the user might receive a blank page.